prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctions.kt

```diff
@@ -319,4 +355,97 @@ object CustomFHIRFunctions {
         }
         return if (type != null) mutableListOf(StringType(type)) else mutableListOf()
     }
+
+    /**
+     * Get the LOINC Code from the LIVD table based on the device id, equipment model id, test kit name id, or the
+     * element model name
+     * @return a list with one value denoting the LOINC Code, or an empty list
+     */
+    fun livdTableLookup(
+        focus: MutableList<Base>,
+        parameters: MutableList<MutableList<Base>>?,
+        metadata: Metadata = Metadata.getInstance()
+    ): MutableList<Base> {
+        val lookupTable = metadata.findLookupTable(name = LivdLookup.livdTableName)
+
+        if (focus.size != 1) {
+            error("Must call the livdTableLookup function on a single observation")
+        }
+
+        val observation = focus.first()
+        if (observation !is Observation) {
+            error("Must call the livdTableLookup function on an observation")
+        }
+
+        var result: String? = ""
+        if (observation is Observation) {
+            // Maps to OBX 17 CWE.1 Which is coding[1].code
+            val coding = (observation as Observation?)?.code?.coding
+            if (!coding.isNullOrEmpty()) {
+                coding[0]?.code?.let {
+                    result = LivdLookup.find(
```

Comment:
These find functions throw an exception you are not catching if the table name is not right. You need to convert that error to a SchemaException.

Reply:
Not sure I am understanding what this is asking for. You want me to convert the error in find to a Schema Exception and then catch it and handle it here?